File: backend/app/models/menu_upload_job.py
from datetime import datetime
from pydantic import BaseModel, Field
from app.models.enums import JobStatus, UploadType, JobType
from app.models.uploaded_file import UploadedFile

class MenuUploadJob(BaseModel):
    job_id: str
    job_type: JobType = JobType.MENU_UPLOAD
    restaurant_id: str
    upload_type: UploadType
    platform: str = "zomato"
    status: JobStatus = JobStatus.QUEUED
    files: list[UploadedFile] = []
    total_files: int = 0
    raw_text: str | None = None
    proposed_changes: list[dict] = Field(default_factory=list, description="For PRICE_UPDATE jobs, holds the diff.")
    created_at: datetime = Field(
        default_factory=datetime.utcnow
    )

    started_at: datetime | None = None
    completed_at: datetime | None = None
    error: str | None = None

    progress: int | None = None
    step: str | None = None

    chain_outputs: dict = Field(default_factory=dict, description="Outputs from various AI chain steps")

File: frontend/src/model/menu-upload-job.js
import mongoose from "mongoose";

const MenuUploadJobSchema = new mongoose.Schema(
    {
        job_id: {
            type: String,
            required: true,
            unique: true,
            index: true,
        },
        job_type: {
            type: String,
            default: "menu_upload",
        },
        restaurant_id: {
            type: String,
            required: true,
            index: true,
        },
        upload_type: {
            type: String,
            default: "images",
        },
        platform: {
            type: String,
            default: "zomato",
            index: true,
        },
        status: {
            type: String,
            default: "queued",
            index: true,
        },
        files: {
            type: Array,
            default: [],
        },
        total_files: {
            type: Number,
            default: 0,
        },
        raw_text: {
            type: String,
            default: null,
        },
        proposed_changes: {
            type: Array,
            default: [],
        },
        progress: {
            type: Number,
            default: 0,
        },
        step: {
            type: String,
            default: null,
        },
        error: {
            type: String,
            default: null,
        },
        chain_outputs: {
            type: mongoose.Schema.Types.Mixed,
            default: {},
        },
        created_at: {
            type: Date,
            default: Date.now,
        },
        started_at: {
            type: Date,
            default: null,
        },
        completed_at: {
            type: Date,
            default: null,
        },
    },
    {
        collection: "menu_upload_jobs",
        timestamps: { createdAt: "created_at", updatedAt: "updated_at" },
        strict: false,
    }
);

export default mongoose.models.MenuUploadJob ||
    mongoose.model("MenuUploadJob", MenuUploadJobSchema);
